helix-core/src/test/java/org/apache/helix/TestHelper.java

```diff
@@ -794,7 +797,12 @@ public static boolean verify(Verifier verifier, long timeout) throws Exception {
     long start = System.currentTimeMillis();
     do {
       boolean result = verifier.verify();
-      if (result || (System.currentTimeMillis() - start) > timeout) {
+      long now = System.currentTimeMillis();
+      if (result || (now - start) > timeout) {
```

Comment:
This code looks chaotic. Please simplify.

boolean result = verifier.verify();
boolean timeout = (System.currentTimeMillis() - start) > timeout;
if (result || timeout) {
  if (timeout) {
    LOG.error()
  }
  return result;
}

Reply:
changed.